app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala

```diff
@@ -75,19 +80,66 @@ class SchemeMembersListController @Inject()(
       form(manualOrUpload, membersDetails.size >= maxSchemeMembers)
         .bindFromRequest()
         .fold(
-          formWithErrors =>
-            BadRequest(
-              view(formWithErrors, viewModel(srn, page, manualOrUpload, mode, membersDetails.map(_.fullName)))
-            ),
+          formWithErrors => {
+            filterDeletedMembers(srn).map { filteredMembers =>
+              val memberNames = filteredMembers.map { case (_, details, _) => details.fullName }
+              BadRequest(
+                view(
+                  formWithErrors,
+                  viewModel(srn, page, manualOrUpload, mode, memberNames)
+                )
+              )
+            }
+          }.merge,
           value => {
-            if (membersDetails.length == maxSchemeMembers && value) {
+            if (membersDetails.size == maxSchemeMembers && value) {
               Redirect(routes.HowToUploadController.onPageLoad(srn))
             } else {
-              Redirect(navigator.nextPage(SchemeMembersListPage(srn, value, manualOrUpload), mode, request.userAnswers))
+              Redirect(
+                navigator.nextPage(SchemeMembersListPage(srn, value, manualOrUpload), mode, request.userAnswers)
+              )
             }
           }
         )
     }
+
+  // merges member details with member states and filters out any soft deleted members
+  private def filterDeletedMembers(
```

Comment:
Struggling to understand why do we need to filter out on the validation error display, i.e. why is this method called from onSubmit - formWithErrors part.

Reply:
I forgot to add to on page load aswell - the page should not display any deleted members